src/core/shaderlib/lighting/lighting.js

```diff
@@ -20,9 +20,68 @@
 
 import lightingShader from './lighting.glsl';
 import project from '../project/project';
+import {COORDINATE_SYSTEM} from '../../lib/constants';
+import {projectPosition} from '../project/project-functions';
 
 export default {
   name: 'lighting',
   dependencies: [project],
-  vs: lightingShader
+  vs: lightingShader,
+  getUniforms
 };
+
+const INITIAL_MODULE_OPTIONS = {};
+
+const DEFAULT_LIGHTS_POSITION = [-122.45, 37.75, 8000];
+const DEFAULT_LIGHTS_STRENGTH = [2.0, 0.0];
+const DEFAULT_AMBIENT_RATIO = 0.05;
+const DEFAULT_DIFFUSE_RATIO = 0.6;
+const DEFAULT_SPECULAR_RATIO = 0.8;
+
+function getUniforms(opts = INITIAL_MODULE_OPTIONS) {
+  if (!opts.lightSettings) {
+    return {};
+  }
+
+  const {
+    numberOfLights = 1,
```

Comment:
For another PR: It would seem nice to have an API where we take an array of light descriptors rather than have to fiddle with three parameters?

```
lightedModel.setModuleParameters({lights: [
   new Light({position, intensity]),
   new Light(),
   new Light(),
]});
```

This would be particularly interesting if we can find a way to move (parts of?) this module to luma.gl.